pandas/_libs/tslibs/offsets.pyx

```diff
@@ -4762,6 +4763,13 @@ cpdef to_offset(freq, bint is_period=False):
                         f"instead of \'{name}\'"
                     )
                 elif is_period is True and name in c_OFFSET_DEPR_FREQSTR:
+                    if name.startswith("A"):
+                        warnings.warn(
+                            f"\'{name}\' will be deprecated, please use "
+                            f"\'{c_DEPR_ABBREVS.get(name)}\' instead.",
+                            FutureWarning,
+                            stacklevel=find_stack_level(),
+                        )
```

Comment:
just for my understanding, why is `if name.startswith("A")` necessary?

Reply:
it's necessary in order to show the warning  `FutureWarning: 'A' will be deprecated, please use 'Y' instead.` when we pass "A" as frequency: `pd.period_range(freq="A", start="1/1/2001", end="12/1/2009")`

In case is_period is True and freq is in OFFSET_DEPR_FREQSTR we convert frequency silently. After adding frequency "A" to OFFSET_DEPR_FREQSTR we convert "A" to "Y-DEC" silently, e.g.
```
>>> pd.period_range(freq="A", start="1/1/2001", end="12/1/2005")
PeriodIndex(['2001', '2002', '2003', '2004', '2005'], dtype='period[Y-DEC]')
```
without showing warning, but I think we should.

Reply:
ok, so we need this extra one because here we're doing both the A->Y renaming, and Y-YE?

should the warning already just say

> 'A' will be deprecated, please use 'YE' instead

then?

Reply:
but would it be correct? This is period and  frequency should be 'Y' not 'YE'.

Reply:
you're right, thanks, this is for period